src/plugins/map_generation/src/systems/spawn_map_agents.rs

```
@@ -0,0 +1,212 @@
+use crate::{
+	components::map::{
+		agents::{AgentsLoaded, Enemy, Player},
+		cells::{MapCells, agent::Agent},
+	},
+	grid_graph::grid_context::{GridContext, GridDefinition, GridDefinitionError},
+	traits::{GridCellDistanceDefinition, grid_min::GridMin},
+};
+use bevy::prelude::*;
+use common::traits::{thread_safe::ThreadSafe, try_insert_on::TryInsertOn};
+
+impl<TCell> MapCells<Agent<TCell>>
+where
+	TCell: GridCellDistanceDefinition + ThreadSafe,
+{
+	pub(crate) fn spawn_map_agents(
+		mut commands: Commands,
+		cells: Query<(Entity, &Self)>,
+	) -> Result<(), Vec<GridDefinitionError>> {
+		let mut errors = vec![];
+
+		for (entity, MapCells { cells, size, .. }) in &cells {
+			let context = GridContext::try_from(GridDefinition {
+				cell_count_x: size.x,
+				cell_count_z: size.z,
+				cell_distance: TCell::CELL_DISTANCE,
+			});
+			let Ok(context) = context else {
+				errors.push(GridDefinitionError::CellCountZero);
+				continue;
+			};
```

probably should return the error from `GridContext::try_from()` directly.